api/lock_api.go

```diff
@@ -144,15 +142,8 @@ type Committer struct {
 	Email string `json:"email"`
 }
 
-// CurrentCommitter returns a Committer instance populated with the same
-// credentials as would be used to author a commit. In particular, the
-// "user.name" and "user.email" configuration values are used from the
-// config.Config singleton.
-func CurrentCommitter() Committer {
```

Comment:
I like the idea of using `Unmarshal` here. I want to get away from importing `config` on every package. There's obviously a lot of work to do in our codebase, but I think the new `locking` package should start off on the right foot.

```go
package main

import (
  "fmt"

  "github.com/git-lfs/git-lfs/config"
  "github.com/git-lfs/git-lfs/locking"
)

func main() {
  cfg := config.New()
  lockclient := locking.New()
  cfg.Unmarshal(lockclient)
  cfg.Unmarshal(&lockclient.CurrentCommitter)

  fmt.Printf("%+v\n", lockclient)
}